pkg/network/sriov/sriov.go

```diff
@@ -19,4 +19,77 @@
 
 package sriov
 
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
 const AliasPrefix = "sriov-"
+
+func CreateNetworkPCIAnnotationValue(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (string, error) {
+	networkPciMap, err := mapNetworkNameToPCIAddress(networks, interfaces, networkStatusAnnotationValue)
+	if err != nil {
+		return "", err
+	}
+
+	if len(networkPciMap) == 0 {
+		return "", fmt.Errorf("expecting non-emtpy network-pci-map")
+	}
+	networkPciMapBytes, err := json.Marshal(networkPciMap)
+	if err != nil {
+		return "", err
+	}
+
+	return string(networkPciMapBytes), nil
+}
+
+func mapNetworkNameToPCIAddress(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (map[string]string, error) {
+	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	if err != nil {
+		return nil, err
+	}
+	networkNameScheme := namescheme.CreateNetworkNameScheme(networks)
+
+	networkPciMap := map[string]string{}
+	for _, sriovIface := range vmispec.FilterSRIOVInterfaces(interfaces) {
+		multusInterfaceName := networkNameScheme[sriovIface.Name]
+		networkStatusEntry, exist := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
+		if !exist {
+			return nil, fmt.Errorf("failed to find network-status entry with interface %s", multusInterfaceName)
+		}
+		if networkStatusEntry.DeviceInfo == nil || networkStatusEntry.DeviceInfo.Pci == nil {
+			return nil, fmt.Errorf("failed to find device-info/pci-address in network-status annotation for SR-IOV interface %s", sriovIface.Name)
+		}
+
+		pciAddress := networkStatusEntry.DeviceInfo.Pci.PciAddress
+		if pciAddress == "" {
+			return nil, fmt.Errorf("failed to associate pci-address to SRIOV interface %s", sriovIface.Name)
+		}
+		networkPciMap[sriovIface.Name] = pciAddress
+	}
+	return networkPciMap, nil
+}
+
+func mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue string) (map[string]networkv1.NetworkStatus, error) {
+	if len(networkStatusAnnotationValue) == 0 {
+		return nil, fmt.Errorf("network-status annotation is not present")
+	}
+	var networkStatusList []networkv1.NetworkStatus
+	err := json.Unmarshal([]byte(networkStatusAnnotationValue), &networkStatusList)
+	if err != nil {
```

Comment:
```suggestion
	if err := json.Unmarshal([]byte(networkStatusAnnotationValue), &networkStatusList); err != nil {
```

Reply:
I find this format less appealing. It shortens the lines but also lengthens the it, do you care to explain why this is preferred?

Reply:
It limits the `err` variable's scope to the `if` statement.

Reply:
fixed.